runtime/android/core/src/org/xwalk/core/XWalkCoreWrapper.java

```diff
@@ -145,6 +152,11 @@ private boolean findSharedCore() {
         XWalkApplication application = XWalkApplication.getApplication();
         if (application == null) Assert.fail("Must use or extend XWalkApplication");
 
+        if (XWalkSdkVersion.VERIFY_XWALK_APK &&
+                !verifyPackageSignature(XWALK_APK_PACKAGE, XWalkSdkVersion.XWALK_APK_FINGERPRINT)) {
+            return false;
```

Comment:
Agreed: if the signature check fails, you need to do something big and obvious so that the user knows they have an invalid crosswalk apk installed and that's why things aren't running.